### PR TITLE
ISSUE-57: Enabled adding of custom headers to email sending

### PR DESCRIPTION
**Pull request by Svenskapojkarna**
Implements #57

### CODE REVIEW

File: Frends.Community.Email/Definitions.cs [hunk 1]
```diff
@@ -157,6 +157,27 @@ public class Options
         [DefaultValue("\"\"")]
         [UIHint(nameof(SkipAuthentication), "", false)]
         public string Password { get; set; }
+
+        /// <summary>
+        ///     Custom headers for email sending.
```
Comment (jefim) [inline Frends.Community.Email/Definitions.cs:162]: Please remove leading spaces here and in other comments.

File: Frends.Community.Email/EmailTask.cs [hunk 2]
```diff
@@ -309,13 +309,20 @@ private static MimeMessage CreateMimeMessage([PropertyTab] Input message)
...
+            foreach (var ccRecipient in ccRecipients)
+                mail.Cc.Add(MailboxAddress.Parse(ccRecipient));
 
             // Add BCC recipients.
-            foreach (var bccRecipient in bccRecipients) mail.Bcc.Add(MailboxAddress.Parse(bccRecipient));
+            foreach (var bccRecipient in bccRecipients)
+                mail.Bcc.Add(MailboxAddress.Parse(bccRecipient));
+
+            if (headers != null && headers.Length > 0)
+                foreach (var header in headers)
```
Comment (jefim) [inline Frends.Community.Email/EmailTask.cs:324]: In all honesty, this nested if-foreach without any brackets looks really dangerous if someone would ever to add more code and the readability is somewhat flaky to my eye. I beg you to consider adding at least brackets for the `if` statement.

File: README.md [hunk 3]
```diff
@@ -76,16 +76,23 @@ Settings for included attachments.
...
+| Username           | string   | Use this username to log in to the SMTP server.                                                                                                                             | user                     |
+| Password           | string   | Use this password to log in to the SMTP server.                                                                                                                             | password                 |
+| CustomHeaders      | Header[] | Array of custom headers which will be added to sending of email. Can be left empty if no custom headers are needed. See Header.                                             | Check Header             |
+
+### Header
+
+| Property | Type   | Description                 | Example                   |
+|----------|--------|-----------------------------|---------------------------|
+| Key      | String | Key of the custom header.   | Content-Transfer-Encoding |
```
Comment (jefim) [inline README.md:94]: In previous docs you have `string`, but in this place you have `String`. I think it is better to use same style everywhere.

**Comment by Svenskapojkarna** [comment]
PR not updating after push, so closing this one and creating a new one.

**Pull request merged**